### ISSUE TITLE
Thank you for your excellent work，I have some questions about this experimental analysis.

### ISSUE THREAD

**Issue by CCCCCCCABC** [zhjgao/difformer#5]
1.Baseline models such as DiffuSeq do not use the enhancement noise factor F during fast sampling, so they do not set K as low as your 20 during sampling. Did you set them equally to K=20 during the comparison process or did you leave the original model unchanged?

2.The score without the noise factor in the ablation experiment was only 2.85, so why is there such a huge gap?Does this cause include the factors present in the first question?

3.in 5.3 Ablation Experiments section, Figure 5 is a bit unclear to me.The dashed part is where you feed pure Gaussian noise to the denoiser at each diffusion step so that it predicts X0. The solid part is where you feed Gaussian noise only at T = 2000 steps and then iterate sequentially to get X0 each time?

Thank you!

**Comment by zhjgao** [comment]
Hi, thanks for your interest in our work, and sorry for the late response.
1. We leave the number of sampling steps for baselines unchanged.
2. We have further tuned the hyper-parameters under this condition, and this value should be updated to 21.46.
3. The dashed lines in Fig. 5 are like what you said. The solid lines are drawn by evaluating predicted z0_hat at each t during a whole generation process.

**Comment by CCCCCCCABC** [comment]
> Hi, thanks for your interest in our work, and sorry for the late response.
> 
> 1. We leave the number of sampling steps for baselines unchanged.
> 2. We have further tuned the hyper-parameters under this condition, and this value should be updated to 21.46.
> 3. The dashed lines in Fig. 5 are like what you said. The solid lines are drawn by evaluating predicted z0_hat at each t during a whole generation process.


Hello author, I have a few questions to ask you, thank you！
1.Whether the layer normalization directly uses the functions on fairseq's default Embedding？this is -> nn.init.normal_(self.word_embedding.weight , mean=0, std=self.in_channels ** -0.5)
2.Are parameters shared between the Word embedding and logits layers?
3.When I use your anchoring loss function, grid_norm starts with inf, then goes down, at some stage it gets big again, then goes down again, then gets big again, and then goes up all the way, whereas with round_loss it doesn't, can you give me some tips? thank you.

I am looking forward to your reply

**Comment by zhjgao** [comment]
1. The implementation of the embedding normalization lies here https://github.com/zhjgao/difformer/blob/62c7dd732a0494732847fbc21607366510114e5b/difformer/modules.py#L85
2. Yes, we tie the parameters of the embeddings and the output layer.
3. We have not encountered the phenomenon you mentioned. Modifications to `--lr`, `--max-tokens`, and `--clip-norm` may be helpful.
